node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -371,68 +371,79 @@ pub mod standard {
+        supplied_version: &str,
+        descriptor: NodeDescriptor,
+    ) -> Result<NodeDescriptor, ParamError> {
+        match descriptor.node_addr_opt.is_some(){
+            true => Ok(descriptor),
+            false => Err(ParamError::new(
+                "neighbors",&format!("Neighbors supplied without ip addresses and ports aren't valid: '{}--NA--:--NA--",
+                                     if supplied_version.ends_with("@:") || supplied_version.ends_with("::")
+                                     {supplied_version.strip_suffix(':').expect("logic failed")}
+                                     else {supplied_version}))
+            )
+        }
+    }
+
     pub fn convert_ci_configs(
         multi_config: &MultiConfig,
     ) -> Result<Option<Vec<NodeDescriptor>>, ConfiguratorError> {
+        type NodeDescriptorParsingResult = Result<NodeDescriptor, ParamError>;
+
         match value_m!(multi_config, "neighbors", String) {
             None => Ok(None),
             Some(joined_configs) => {
                 let cli_configs: Vec<String> = joined_configs
                     .split(',')
                     .map(|s| s.to_string())
                     .collect_vec();
-                if cli_configs.is_empty() {
-                    Ok(None)
-                } else {
-                    let dummy_cryptde: Box<dyn CryptDE> = {
-                        if value_m!(multi_config, "fake-public-key", String) == None {
-                            Box::new(CryptDEReal::new(DEFAULT_CHAIN_ID))
-                        } else {
-                            Box::new(CryptDENull::new(DEFAULT_CHAIN_ID))
-                        }
-                    };
-                    let chain_name = value_m!(multi_config, "chain", String)
-                        .unwrap_or_else(|| DEFAULT_CHAIN_NAME.to_string());
-                    let results = cli_configs
+                let dummy_cryptde: Box<dyn CryptDE> = {
+                    if value_m!(multi_config, "fake-public-key", String) == None {
+                        Box::new(CryptDEReal::new(DEFAULT_CHAIN_ID))
+                    } else {
+                        Box::new(CryptDENull::new(DEFAULT_CHAIN_ID))
+                    }
+                };
+                let chain_name = value_m!(multi_config, "chain", String)
+                    .unwrap_or_else(|| DEFAULT_CHAIN_NAME.to_string());
+                let results = cli_configs
                         .into_iter()
                         .map(
-                            |s| match NodeDescriptor::from_str(dummy_cryptde.as_ref(), &s) {
+                            |string_record| match NodeDescriptor::from_str(dummy_cryptde.as_ref(), &string_record) {
                                 Ok(nd) => match (chain_name.as_str(), nd.mainnet) {
-                                    (DEFAULT_CHAIN_NAME, true) => Ok(nd),
+                                    (DEFAULT_CHAIN_NAME, true) => validate_mandatory_node_addr(&string_record,nd),
```

Comment:
I'm not sure why rustfmt didn't add a space after the comma before nd.

Reply:
It's probably optional at too short names...well, I don't know either...but adding it was exactly as fine for rustfmt as when I had it without.